hdfs-commons/src/main/java/org/apache/mesos/hdfs/config/HdfsFrameworkConfig.java

```diff
@@ -308,4 +312,21 @@ public String getLdLibraryPath() {
   public String getJreVersion() {
     return getConf().get("mesos.hdfs.jre-version", "jre1.7.0_76");
   }
+  
+  public Map<String, String> getMesosSlaveConstraints() {
+    String constraints = getConf().get("mesos.hdfs.constraints");
+    Map<String, String> constraintsMap = new HashMap<String, String>();
+    if (!StringUtils.isBlank(constraints)) {
+      String[] constraintsPairs = constraints.split(";");
+      for (String pair : constraintsPairs) {
+        String[] keyValue = pair.split(":");
+        String key = keyValue[0];
+        String value = keyValue.length == 2 ? keyValue[1]
+            : keyValue.length == 1 ? "" : pair.substring(pair.indexOf(":"));
```

Comment:
It means if  
a:b the key = a , value = b
a then key = a, value = ""
a:b:c:d then key = a and value = b:c:d